factiondraft.ttslua

```diff
@@ -0,0 +1,807 @@
+         tTemp[i] = nil
+         if(  validateSetup(tTemp, currentReach+v.faction.reachVal, playerCount - selectedCount-1) ) then
+            table.insert(pickable, v)
+         end
+      end
+   end
+
+   
+   for i, v in pairs(pickable) do
+      if v ~= nil then
+         self.UI.setAttribute("btn-faction-"..v.id, "color", v.faction.color)
+         self.UI.setAttribute("btn-faction-"..v.id, "interactable", true)
+      end
+   end
+end
+
+function validateSetup(possible, reach, playersLeft)
+   local playerCount = #selectionOrder
+   local minReach = reachMins[playerCount]
+
+   if reach >= minReach then
+      return true
+   end
+
+   for i, v in pairs(possible) do
+      if(v ~= nil) then
+         local tTemp = sCopy(possible)
+         if(playersLeft <= 0) then
+            return false
+         end
+
+         local reachTmp = reach + v.faction.reachVal
+
+         if(reachTmp >= minReach) then
+            return true
+         end
+
+         tTemp[i] = nil
+         if( validateSetup(tTemp, reachTmp, playersLeft-1) ) then
+            return true
+         else
+            possible[i] = nil
+         end
+      end
+   end
+end
+
+function selectFaction(player, value, elem)
+   
+   local nextID = ""
+   local nextName = ""
+   local currentOrderIndex = -1
+   for i, c in ipairs(selectionOrder) do
+      if selections[c.id] == nil then
+         nextID = c.id
+         nextName = c.name
+         currentOrderIndex = i
+         break
+      end
+   end
+
+
+   if player.steam_id ~= nextID and #(tostring(nextID)) > 1 then
+      broadcastToColor("Not your turn yet", player.color)
+      return
+   end
+
+   local faction = factions[tonumber(value)]
+   if testPlayers then
+      selections[nextID] = faction
+   else
+      selections[player.steam_id] = faction
+   end
+   currentReach = currentReach + faction.reachVal
+
+   self.UI.setAttribute("reach-counter", "text", "Reach: " .. currentReach .. "/" .. reachMins[#selectionOrder])
+   self.UI.setAttribute("reach-counter", "color", "#dcdccc")
+
+   self.UI.hide("btn-faction-"..value)
+   local bgColor = self.UI.getAttribute("btn-faction-"..value, "color")
+   local fgColor = self.UI.getAttribute("btn-faction-"..value, "textColor")
+   local text = self.UI.getAttribute("btn-faction-"..value, "text")
+   self.UI.setAttribute("btn-faction-"..value, "color", fgColor)
+   self.UI.setAttribute("btn-faction-"..value, "text", text .. "\n" .. nextName .. " (" .. currentOrderIndex .. ")")
+   self.UI.setAttribute("btn-faction-"..value, "textColor", bgColor)
+   self.UI.setAttribute("btn-faction-"..value, "interactable", false)
+   self.UI.show("btn-faction-"..value)
+   
+   table.insert(turnOrder, 1, faction.color)
+
+   if value == "4" then
+      self.UI.show("btn-faction-8")
+   end
+
+   local playmat = getNextPlaymat()
+   
+   Wait.frames(
+      function()
+         if playmat == nil then
+            return
+         end
+         playmat.call("spawnRemote", {
```

Comment:
In here you will have to call:
```playmat.UI.setAttribute("subpanel", "active", false)```
To remove the splash screen. Only needed for vagabond selection afterwards.

Reply:
Not sure what you mean... the vagabond selection seems to work as is

Reply:
It does, but it will switch to vagabond slection while the spashscreen is still showing behind.

Reply:
Ah, I understand now, I'll update that